dev/integration_tests/android_views/lib/motion_event_page.dart

```diff
@@ -0,0 +1,297 @@
+
+const String kEventsFileName = 'touchEvents';
+
+class MotionEventPage extends Page {
+  const MotionEventPage()
+      : super('Place circle', const ValueKey<String>('MotionEventPage'));
+
+  @override
+  Widget build(BuildContext context) {
+    return MotionEventBody();
+  }
+}
+
+/// Wraps a flutter driver [DataHandler] with one that waits until a delegate is set.
+///
+/// This allows the driver test to call [FlutterDriver.requestData] before the handler was
+/// set by the app in which case the requestData call will only complete once the app is ready
+/// for it.
+class FutureDataHandler {
+  final Completer<DataHandler> handlerCompleter = Completer<DataHandler>();
+
+  Future<String> handleMessage(String message) async {
+    final DataHandler handler = await handlerCompleter.future;
+    return handler(message);
+  }
+}
+
+FutureDataHandler driverDataHandler = FutureDataHandler();
+
+class MotionEventBody extends StatefulWidget {
+  @override
+  State createState() => MotionEventBodyState();
+}
+
+class MotionEventBodyState extends State<MotionEventBody> {
+  static const int kEventsBufferSize = 1000;
+
+  MethodChannel viewChannel;
+
+  /// The list of motion events that were passed to the FlutterView.
+  List<Map<String, dynamic>> flutterViewEvents = <Map<String, dynamic>>[];
+
+  /// The list of motion events that were passed to the embedded view.
+  List<Map<String, dynamic>> embeddedViewEvents = <Map<String, dynamic>>[];
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        SizedBox(
+          height: 300.0,
+          child: AndroidView(
+              key: const ValueKey<String>('MotionEventPageLoaded'),
+              viewType: 'simple_view',
+              onPlatformViewCreated: onPlatformViewCreated),
+        ),
+        Expanded(
+          child: ListView.builder(
+            itemBuilder: buildEventTile,
+            itemCount: flutterViewEvents.length,
+          ),
+        ),
+        Row(
+          children: <Widget>[
+            RaisedButton(
+              child: const Text('RECORD'),
+              onPressed: listenToFlutterViewEvents,
+            ),
+            RaisedButton(
+              child: const Text('CLEAR'),
+              onPressed: () {
+                setState(() {
+                  flutterViewEvents.clear();
+                  embeddedViewEvents.clear();
+                });
+              },
+            ),
+            RaisedButton(
+              child: const Text('SAVE'),
+              onPressed: () {
+                const StandardMessageCodec codec = StandardMessageCodec();
+                saveRecordedEvents(
+                    codec.encodeMessage(flutterViewEvents), context);
+              },
+            ),
+            RaisedButton(
+              key: const ValueKey<String>('play'),
+              child: const Text('PLAY FILE'),
+              onPressed: () {
```

Comment:
Avoid reformatting the previous code if the previous formatting is good. (in multiple places)